### PR TITLE
feat(skyrim-platform): add consoleMessage event

### PR DESCRIPTION
**Pull request by russo-2025**
usage example:
```
const htmlEscapes: any = {
  '"': '\\"',
  "'": "\\'",
  '\\': '\\\\',
};

const htmlEscaper = /[&<>"'\\\/]/g;

on('consoleMessage', (e: ConsoleMessageEvent) => {
  const msg = e.message.replace(htmlEscaper, (match) => htmlEscapes[match]);
  browser.executeJavaScript('console.log("' + msg + '")');
});
```
image

### CODE REVIEW

File: skyrim-platform/src/platform_se/skyrim_platform/FridaHooks.cpp [hunk 1]
```diff
@@ -105,6 +106,7 @@ void SetupFridaHooks()
   w.Attach(listener, 5367792, RENDER_MAIN_MENU);
   w.Attach(listener, 19244800, SEND_EVENT);
   w.Attach(listener, 19245744, SEND_EVENT_ALL);
+  w.Attach(listener, 0x85C423, CONSOLE_VPRINT);
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/FridaHooks.cpp:109]: Please make addresses look the same (all hex or all decimal)

File: skyrim-platform/src/platform_se/skyrim_platform/EventsApi.cpp [hunk 2]
```diff
@@ -459,6 +459,16 @@ void EventsApi::IpcSend(const char* systemName, const uint8_t* data,
   SendEvent("ipcMessage", { JsValue::Undefined(), ipcMessageEvent });
 }
 
+void EventsApi::SendConsoleMsgEvent(char* msg_)
+{
+  std::string msg(msg_);
+  SkyrimPlatform::GetSingleton().AddUpdateTask([=] {
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventsApi.cpp:465]: Please change to `AddTickTask` to work in the main menu
Or tell me why it isn't possible

File: skyrim-platform/src/platform_se/skyrim_platform/EventsApi.cpp [hunk 3]
```diff
@@ -459,6 +459,16 @@ void EventsApi::IpcSend(const char* systemName, const uint8_t* data,
   SendEvent("ipcMessage", { JsValue::Undefined(), ipcMessageEvent });
 }
 
+void EventsApi::SendConsoleMsgEvent(char* msg_)
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventsApi.cpp:462]: please change to `const char *`

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Review by Pospelove — APPROVED** [review]

File: skyrim-platform/src/platform_se/skyrim_platform/FridaHooks.cpp [hunk 4]
```diff
@@ -42,7 +42,8 @@ enum _ExampleHookId
   APPLY_MASKS_TO_RENDER_TARGET,
   RENDER_MAIN_MENU,
   SEND_EVENT,
-  SEND_EVENT_ALL
+  SEND_EVENT_ALL,
+  CONSOLE_VPRINT
```
Comment (nic11) [inline skyrim-platform/src/platform_se/skyrim_platform/FridaHooks.cpp:46]: @Pospelove BTW I think it's better to add trailing comma in such cases. Modern C++ should allow that, if I remember correctly

Comment (Pospelove) [reply skyrim-platform/src/platform_se/skyrim_platform/FridaHooks.cpp:46]: @nic11 ok

**Pull request merged**